x/mongo/driver/topology/connection.go

```diff
@@ -108,7 +109,19 @@ func (c *connection) connect(ctx context.Context) {
 	}
 	defer close(c.connectDone)
 
+	c.mu.Lock()
 	ctx, c.cancelConnectContext = context.WithCancel(ctx)
+	c.mu.Unlock()
+
+	defer func() {
+		c.mu.Lock()
```

Comment:
nit: Can we do `defer c.mu.Unlock()` right below this instead of unlocking after all of the logic is done? It's not a big deal now, but I don't want someone to make a change in the future that causes us to exit this defer at an earlier point and leave without releasing the lock. You could add a similar defer to the `closeConnectContext` function.

Reply:
done